controllers/backplaneconfig_controller.go

```diff
@@ -1038,6 +1038,13 @@ func (r *MultiClusterEngineReconciler) setDefaults(ctx context.Context, m *backp
 		updateNecessary = true
 	}
 
+	// managedserviceaccount preview component upgraded in 2.9.0
+	if m.Prune(backplanev1.ManagedServiceAccountPreview) {
+		// if the preview was pruned, enable the non-preview version instead
```

Comment:
This sounds like it will always enable managedSA on an upgrade since msa-preview will always be there and need pruning. It should only enable managedSA if msa-preview was also enabled.